Python/Product/Analysis/Interpreter/Ast/AstPythonMultipleModules.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.PythonTools.Analysis;
+using Microsoft.PythonTools.Analysis.Infrastructure;
+
+namespace Microsoft.PythonTools.Interpreter.Ast {
+    class AstPythonMultipleModules : IPythonModule, IPythonMultipleMembers, ILocatedMember {
+        private IList<IPythonModule> _modules;
+        private IModuleContext _importContext;
+        private bool _imported;
+
+        public AstPythonMultipleModules() {
+            _modules = Array.Empty<IPythonModule>();
+        }
+
+        private AstPythonMultipleModules(IPythonModule[] members) {
+            _modules = members;
+        }
+
+        public AstPythonMultipleModules(IEnumerable<IPythonModule> members) {
+            _modules = members.ToArray();
+        }
+
+        public static IPythonModule Combine(IPythonModule x, IPythonModule y) {
+            if (x == null && y == null) {
+                throw new InvalidOperationException("Cannot add two null members");
+            } else if (x == null || (x.MemberType == PythonMemberType.Unknown && !(x is ILazyMember))) {
+                return y;
+            } else if (y == null || (y.MemberType == PythonMemberType.Unknown && !(y is ILazyMember))) {
+                return x;
+            } else if (x == y) {
+                return x;
+            }
+
+            var mmx = x as AstPythonMultipleModules;
+            var mmy = y as AstPythonMultipleModules;
+
+            if (mmx != null && mmy == null) {
+                mmx.AddModule(y);
+                return mmx;
+            } else if (mmy != null && mmx == null) {
+                mmy.AddModule(x);
+                return mmy;
+            } else if (mmx != null && mmy != null) {
+                mmx.AddModules(mmy._modules);
+                return mmx;
```

Comment:
Does it matter anywhere that we return `mmx` instead of `mmy`?

Reply:
No. This should only be being used while we're constructing objects and before the analyzer ever gets a hold of it (great question though)